sherpa/astro/ui/tests/test_astro_ui.py

```diff
@@ -428,7 +429,11 @@ def save_arrays(self):
         rtol = 0
         atol = 1e-5
         self.assertIsInstance(out, Data1D)
-        self.assertEqual(out.name, ofh.name, msg="file name")
+
+        # remove potential dm syntax introduced by backend before checking for equality
+        out_name = re.sub("\[.*\]", "", out.name)
```

Comment:
I'm happy leaving the code as-is, as there's a comment explaining why the line is needed.